Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Services/GlobalSettingsService.cs

```diff
@@ -0,0 +1,86 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using Amazon.Lambda.TestTool.Models;
+
+namespace Amazon.Lambda.TestTool.Services;
+
+/// <summary>
+/// A service for accessing <see cref="GlobalSettings"/>.
+/// </summary>
+/// <param name="settingsRepository"><see cref="IGlobalSettingsRepository"/></param>
+/// <param name="logger"><see cref="ILogger{GlobalSettingsService}"/></param>
+public class GlobalSettingsService(
+    IGlobalSettingsRepository settingsRepository,
+    ILogger<GlobalSettingsService> logger) : IGlobalSettingsService, IAsyncDisposable
+{
+    private GlobalSettings _currentSettings = new ();
+    private readonly SemaphoreSlim _settingsLock = new (1, 1); // For thread-safe updates
+
+    /// <inheritdoc/>
+    public async Task LoadSettingsAsync()
+    {
+        await _settingsLock.WaitAsync();
+        try
+        {
+            _currentSettings = await settingsRepository.LoadSettingsAsync();
+            logger.LogInformation("Global settings loaded successfully.");
+        }
+        finally
+        {
+            _settingsLock.Release();
+        }
+    }
+
+    /// <inheritdoc/>
+    public GlobalSettings CurrentSettings
+    {
+        get
+        {
+            // Return a clone to prevent external modification of the internal state
+            return _currentSettings.DeepCopy();
+        }
+    }
+
+    /// <inheritdoc/>
+    public async Task UpdateSettings(Action<GlobalSettings> updateAction)
+    {
+        // Use a lock to ensure thread-safe updates
+        _settingsLock.Wait();
+        try
+        {
+            var newSettings = _currentSettings.DeepCopy(); // Create a mutable copy to update
+            updateAction(newSettings); // Apply changes
+
+            if (!AreSettingsEqual(_currentSettings, newSettings))
+            {
+                _currentSettings = newSettings;
+                logger.LogInformation("Global settings updated.");
+                _ = settingsRepository.SaveSettingsAsync(_currentSettings); // Fire-and-forget save

```

Comment:
You are sort of breaking your lock here by doing a fire and forget. You could exit the lock before `SaveSettingsAsync` completes and then another thread could call the `LoadSettingsAsync` method in this class acquire the lock and attempt to read at the same time the save is happening.

A corner case but not sure if you gain any perceived performance improvement by doing a fire and forget and you are opening it up to other potential issues.